internal/access/access_key.go

```diff
@@ -23,20 +29,31 @@ func ListAccessKeys(c *gin.Context, identityID uid.ID, name string, showExpired
 		ByIssuedForID:  identityID,
 		ByName:         name,
 	}
-	return data.ListAccessKeys(db, opts)
+	return data.ListAccessKeys(rCtx.DBTxn, opts)
 }
 
 func CreateAccessKey(c *gin.Context, accessKey *models.AccessKey) (body string, err error) {
-	db, err := RequireInfraRole(c, models.InfraAdminRole)
-	if err != nil {
-		return "", HandleAuthErr(err, "access key", "create", models.InfraAdminRole)
+	rCtx := GetRequestContext(c)
+
+	if rCtx.Authenticated.AccessKey != nil && !rCtx.Authenticated.AccessKey.Scopes.Includes(models.ScopeAllowCreateAccessKey) {
+		// non-login access keys can not currently create other access keys.
+		return "", fmt.Errorf("%w: cannot use an access key to create other access keys", internal.ErrBadRequest)
```

Comment:
```suggestion
		return "", fmt.Errorf("%w: cannot use an access key not issued from login to create other access keys", internal.ErrBadRequest)
```

This is a bit difficult to word, but here is my shot at it